Sources/App/Repositories/AppUserRepository.swift

```diff
@@ -2,14 +2,15 @@ import Fluent
 import Vapor
 
 protocol AppUserRepository {
-    func all(searchterm: String?, on request: Request) -> EventLoopFuture<Page<AppUser>>
-    func delete(_ user: AppUser) -> EventLoopFuture<Void>
-    func find(_ id: AppUser.IDValue?) -> EventLoopFuture<AppUser?>
-    func save(_ user: AppUser) -> EventLoopFuture<AppUser>
+    func paginateAppUsers(searchterm: String?, on request: Request) -> EventLoopFuture<Page<AppUser>>
+    func deleteAppUser(_ user: AppUser) -> EventLoopFuture<Void>
+    func findAppUser(_ id: AppUser.IDValue?) -> EventLoopFuture<AppUser?>
+    func findAppUserByEmail(_ email: String) -> EventLoopFuture<AppUser?>
+    func saveAppUser(_ user: AppUser) -> EventLoopFuture<AppUser>
 }
 
 extension DatabaseRepository: AppUserRepository {
-    func all(
+    func paginateAppUsers(
         searchterm: String?,
         on request: Request
     ) -> EventLoopFuture<Page<AppUser>> {
```

Comment:
on line 19, should we also add a filter by name now that we have added the field?

and a follow up question since we are using PostgreSQL:
Last I checked `~~` gets parsed as `LIKE` which means the search is case sensitive on postgres. Should we add something like 
```swift
import FluentKit

extension QueryBuilder {
    @discardableResult
    func caseInsensitiveContains<Joined, Field>(
        _ schema: Joined.Type,
        _ field: KeyPath<Joined, Field>,
        _ value: Field.Value
    ) -> Self
        where Field: QueryableProperty, Field.Model == Joined, Joined: Schema
    {
        filter(Joined.path(for: field), .custom("ILIKE"), "%\(value)%")
    }

    @discardableResult
    func caseInsensitiveContains<Field>(
        _ field: KeyPath<Model, Field>,
        _ value: Field.Value
    ) -> Self
        where Field: QueryableProperty, Field.Model == Model
    {
        filter(Model.path(for: field), .custom("ILIKE"), "%\(value)%")
    }
}
```
to the project as well, or maybe at least add a comment to the method, so any developers that are used to the behaviour of MySQL don't expect PostgreSQL to act the same way...